### PR TITLE
Sscs 2774 ref data domain

### PR DESCRIPTION
**Pull request by sathishbabuj**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            XMLStreamReader reader =
+                xmlInputFactory.createXMLStreamReader(gaps2RefDataInputStream);
+
+            while (reader.hasNext()) {
+                int event = reader.next();
+
+                switch (event) {
+                    case XMLStreamConstants.START_ELEMENT:
+                        if (APPEAL_STATUS.equals(reader.getLocalName())) {
+                            appealStatus = new AppealStatus();
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java]: If you cannot think of a way to improve this warning in codacy, we can easily ignore that issue in codacy. No need to remove codacy at all.

File: src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import javax.xml.stream.XMLStreamException;
+
+import org.junit.Before;
+import org.junit.Test;
+import uk.gov.hmcts.reform.sscs.models.refdata.AppealStatus;
+
+public class AppealStatusReferenceDataLoaderTest {
+
+    public static final String GAPS2_REFERENCE_DATA_XML =
+        "reference-data/SSCS_Extract_ReferenceTest_2017-06-30-09-01-31.xml";
```
Comment (iguannago) [inline src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java]: Why are you using a different Delta xml file? What's wrong with the existing one? I think we should use the same Delta file to keep things simple.

Comment (sathishbabuj) [reply src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java]: Please read the code carefully. Its for reference data and not delta.

File: src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java (hunk 1 above)
Comment (sathishbabuj) [reply src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java]: Its good to follow the QA checks recommended by the reforms program. No need to add extra checks.

File: src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java (hunk 2 above)
Comment (iguannago) [reply src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java]: sorry, I mean a  different Reference file?

Comment (sathishbabuj) [reply src/test/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoaderTest.java]: The complete reference file is 2.7MB. Not need to use that big file for unit tests

File: src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java (hunk 1 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java]: codacy is recommended in the program, Sathish.

**Pull request merged**